pandas/core/dtypes/missing.py

```diff
@@ -606,15 +607,19 @@ def is_valid_nat_for_dtype(obj, dtype: DtypeObj) -> bool:
     if not lib.is_scalar(obj) or not isna(obj):
         return False
     if dtype.kind == "M":
-        return not isinstance(obj, np.timedelta64)
+        return not isinstance(obj, (np.timedelta64, Decimal))
```

Comment:
this is really strange that you need to do this

Reply:
can you just test dtype.kind == 'O' first?

Reply:
no bc dtype.kind == "O" includes Period and Interval